app/controllers/superpowers_controller.rb

```diff
@@ -53,4 +52,10 @@ def destroy
   def superpower_params
     params.require(:superpower).permit(:name, :description)
   end
+
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      raise ActionController::RoutingError.new("Not Found")
```

Comment:
consider redirecting the user and showing a flash message that they tried to access something they're unauthorized to do instead of raising an exception

Reply:
also I would make the `authorize_user` method a `protected` or `private` method